File: src/AILogistics.Domain/Entities/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AILogistics.Domain.Entities
{
    public class Customer : BaseEntity
    {
        public string CompanyName { get; set; }
        public string ContactPerson { get; set; }
        public string? Email {  get; set; }
        public string PhoneNumber { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State {  get; set; }

        public string Country { get; set; }
        public string PostalCode { get; set; }
        public bool IsActive { get; set; }

        public ICollection<Shipment> Shipments { get; set; } = new List<Shipment>();


    }
}

File: Client/src/types/customer.ts
export interface Customer extends CustomerBaseModel {
    id: number;
    isActive: boolean;
    rowVersion: string;
}

export interface CreateCustomerRequest extends CustomerBaseModel { }

export interface UpdateCustomerRequest extends CustomerBaseModel {
    rowVersion: string;
}

export interface CustomerBaseModel {
    companyName: string;
    contactPerson: string;
    email: string;
    phoneNumber: string;
    address: string;
    city: string;
    state: string;
    country: string;
    postalCode: string;
}

export const emptyCustomer: CustomerBaseModel = {
    companyName: '',
    contactPerson: '',
    email: '',
    phoneNumber: '',
    address: '',
    city: '',
    state: '',
    country: '',
    postalCode: '',
}

export interface UpdateCustomerRequest  extends CustomerBaseModel{
    rowVersion: string;
}

export type CustomerValidationErrors = Partial<Record<keyof CustomerBaseModel, string>>